client/state/data-layer/wpcom/comments/index.js

```diff
@@ -213,19 +214,21 @@ export const deleteComment = ( { dispatch, getState }, action ) => {
 	);
 };
 
-export const announceDeleteSuccess = ( { dispatch }, { options } ) => {
+export const handleDeleteSuccess = ( { dispatch }, { options, refreshCommentListQuery } ) => {
 	const showSuccessNotice = get( options, 'showSuccessNotice', false );
-	if ( ! showSuccessNotice ) {
-		return;
+	if ( showSuccessNotice ) {
+		dispatch(
+			successNotice( translate( 'Comment deleted permanently.' ), {
+				duration: 5000,
+				id: 'comment-notice',
+				isPersistent: true,
+			} )
+		);
 	}
 
-	dispatch(
-		successNotice( translate( 'Comment deleted permanently.' ), {
-			duration: 5000,
-			id: 'comment-notice',
-			isPersistent: true,
-		} )
-	);
+	if ( ! isEmpty( refreshCommentListQuery ) ) {
```

Comment:
Should we default `refreshCommentListQuery` to `null` or something similar to avoid needing the isEmpty check?